src/aws-cpp-sdk-core/source/internal/AWSHttpResourceClient.cpp

```diff
@@ -79,7 +79,10 @@ namespace Aws
         }
 
         AWSHttpResourceClient::AWSHttpResourceClient(const Aws::Client::ClientConfiguration& clientConfiguration, const char* logtag)
-        : m_logtag(logtag), m_retryStrategy(clientConfiguration.retryStrategy), m_httpClient(nullptr)
+        : m_logtag(logtag),
+          m_userAgent(Aws::Client::ComputeUserAgentString(&clientConfiguration)),
```

Comment:
why do we need to pass `clientConfiguration` as a raw ptr instead of just a normal lvalue ref?

Reply:
Old `ComputeUserAgentString` does no accept any argument.
This is an approach to make it 
1. backward compatible
2. allow client config to be optional (in case this method is called in a place where there is no client config around).